src/components/AppIcon/AppIcon.js

```diff
@@ -78,25 +87,33 @@ const AppIconContent = ({ app, size, src }) => {
       <RemoteIcon src={legacyAppIconUrl(app)} size={size} />
     </RemoteIcon>
   )
-}
+})
 
 // Display a remote icon if found,
-// otherwise a provided fallback if provided,
-// otherwise the default icon.
-const RemoteIcon = ({ src, size, children }) => {
-  const fallback = children || <IconBase size={size} src={iconSvgDefault} />
-  return src === null ? (
-    fallback
-  ) : (
-    <RemoteImage src={src}>
-      {({ exists }) => (exists ? <IconBase size={size} src={src} /> : fallback)}
-    </RemoteImage>
-  )
-}
+// or the provided fallback, or the default icon.
+const RemoteIcon = React.memo(({ src, size, children }) => {
+  const { exists, loading } = useImageExists(src)
+  const [displayFallback, setDisplayFallback] = useState(false)
+
+  useEffect(() => {
+    const timer = setTimeout(
+      () => setDisplayFallback(true),
+      DISPLAY_FALLBACK_DELAY
+    )
+    return () => clearTimeout(timer)
+  }, [])
+
+  // display fallback
+  if ((!exists && !loading) || (loading && displayFallback)) {
```

Comment:
Wonder if we can simply this to

```suggestion
  if (!exists || displayFallback) {
```

Reply:
I don’t think we can:

The `!exists` alone would display the fallback immediately, we need to ensure it has finished loading before (i.e. the image really doesn’t exist).

The `displayFallback` alone would display it all the time, even after the loading has finished, which means the image would never be displayed if it exists.

Reply:
(Note: it could be nice to add this delay mechanism to `useImageExists()`, and we could also ensure that users don’t have to do this kind of complex checks on their side).